File: source/dotnet/source/cmf.bus/cmf.bus.berico/ITransportProvider.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace cmf.bus.berico
{
    public interface ITransportProvider : IDisposable
    {
        event Action<IEnvelopeDispatcher> OnEnvelopeReceived;


        void Send(Envelope env);

        void Register(IRegistration registration);

        void Unregister(IRegistration registration);
    }
}

File: source/java/src/main/java/cmf/bus/berico/ITransportProvider.java
package cmf.bus.berico;

import cmf.bus.Envelope;
import cmf.bus.IRegistration;

public interface ITransportProvider {

	void onEnvelopeReceived(IEnvelopeReceivedCallback callback);
	
	
    void register(IRegistration registration) throws Exception;

    void unregister(IRegistration registration) throws Exception;

    void send(Envelope envelope) throws Exception;
}
